linera-base/src/crypto/secp256k1/evm.rs

```diff
@@ -657,4 +721,22 @@ mod tests {
         let sig2: EvmSignature = serde_json::from_str(&s).unwrap();
         assert_eq!(sig, sig2);
     }
+
+    #[test]
+    fn public_key_recovery() {
+        use crate::crypto::{
+            secp256k1::evm::{EvmKeyPair, EvmPublicKey, EvmSignature},
+            CryptoHash, TestString,
+        };
+        let key_pair = EvmKeyPair::generate();
+        let address = key_pair.public_key.address();
+        let msg = TestString("hello".into());
+        let prehash = CryptoHash::new(&msg);
+        let sig = EvmSignature::new(prehash, &key_pair.secret_key);
+
+        assert!(sig.check_with_recover(&msg, address.0 .0).is_ok());
```

Comment:
I think this would print the error if it fails:

```suggestion
        sig.check_with_recover(&msg, address.0 .0).unwrap();
```

Reply:
Or `assert_matches!(Ok(_), sig.check_with_recover(&msg, address.0 .0))`

Reply:
Actually, they're not equivalent - in my version it won't print what was the `Err(_)` variant.

I'll update.